src/index.ts

```diff
@@ -0,0 +1,283 @@
+		setFailed(
+			`🚨 Wrangler v1 is no longer supported by this action. Please use major version 2 or greater`,
+		);
+	}
+	startGroup("📥 Installing Wrangler");
+	const command = `npm install wrangler@${config["WRANGLER_VERSION"]}`;
+	info(`Running command: ${command}`);
+	execSync(command, { cwd: config["workingDirectory"], env: process.env });
+	info(`✅ Wrangler installed`);
+	endGroup();
+}
+
+function authenticationSetup() {
+	process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_API_TOKEN"];
+	process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_ACCOUNT_ID"];
+}
+
+async function execCommands(commands: string[], cmdType: string) {
+	if (!commands.length) {
+		return;
+	}
+
+	startGroup(`🚀 ${cmdType} Commands Group`);
+
+	const arrPromises = commands.map(async (command) => {
+		const cmd = command.startsWith("wrangler")
+			? `${getNpxCmd()} ${command}`
+			: command;
+
+		info(`🚀 Executing command: ${cmd}`);
+
+		return await execAsync(cmd, {
+			cwd: config["workingDirectory"],
+			env: process.env,
+		});
+	});
+
+	await Promise.all(arrPromises);
+	endGroup();
+}
+
+/**
+ * A helper function to get the secret from the environment variables.
+ */
+function getSecret(secret: string) {
+	if (!secret) {
+		setFailed("No secret provided");
+	}
+
+	const value = process.env[secret];
+	if (!value) {
+		setFailed(`Secret ${secret} not found`);
+	}
+
+	return value;
+}
+
+async function legacyUploadSecrets(
+	secrets: string[],
+	environment?: string,
+	workingDirectory?: string,
+) {
+	try {
+		const arrPromises = secrets
+			.map((secret) => {
+				const command = `echo ${getSecret(
+					secret,
+				)} | ${getNpxCmd()} wrangler secret put ${secret}`;
+				return environment ? command.concat(`--env ${environment}`) : command;
+			})
+			.map(
+				async (command) =>
+					await execAsync(command, {
+						cwd: workingDirectory,
+						env: process.env,
+					}),
+			);
+
+		await Promise.all(arrPromises);
+	} catch {
+		setFailed(`🚨 Error uploading secrets`);
+	}
+}
+
+async function uploadSecrets() {
+	const secrets: string[] = config["secrets"];
+	const environment = config["ENVIRONMENT"];
+	const workingDirectory = config["workingDirectory"];
+
+	if (!secrets.length) {
+		return;
+	}
+	startGroup("🔑 Uploading Secrets");
+
+	if (semverCompare(config["WRANGLER_VERSION"], "3.4.0"))
+		return legacyUploadSecrets(secrets, environment, workingDirectory);
+
+	const secretObj = secrets.reduce((acc: any, secret: string) => {
+		acc[secret] = getSecret(secret);
+		return acc;
+	}, {});
+
+	const environmentSuffix = !environment.length ? "" : ` --env ${environment}`;
+
+	const secretCmd = `echo "${JSON.stringify(secretObj).replaceAll(
+		'"',
+		'\\"',
+	)}" | ${getNpxCmd()} wrangler secret:bulk ${environmentSuffix}`;
+
+	try {
+		execSync(secretCmd, {
+			cwd: workingDirectory,
+			env: process.env,
+			stdio: "ignore",
+		});
+		info(`✅ Uploaded secrets`);
+	} catch {
+		setFailed(`🚨 Error uploading secrets`);
+	}
```

Comment:
just a heads up, as mentioned previously this catch won't be triggered if the secrets fail to upload. we'll need to fix this upstream in wrangler to have it return a non-zero exit code when `wrangler secret:bulk` fails.

Reply:
Nice catch, I will work on that tomorrow in Wrangler!